### ISSUE TITLE
Update ESlinted acronyms

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#4691]
## Feature Description

The list of acronyms that Site Kit enforces casing for should be reviewed and updated to make sure we're covering all the instances that it should be. Offending code is usually references from Google APIs. One such violation that isn't currently covered is `URI`, but there may be others we should add that have slipped by.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The list of acronyms that Site Kit enforces casing for is updated to include the following additional acronyms::
  * `SDK` 
  * `EMM` 
  * `REST` 
  * `GKE` 
  * `URI`  
  * `UX`  
  * `OS` 
  * `GCP` 
  * `DNS`  
  * `FCM`  
  * `IAM` 
  * `IP` 
  * `ISO`

## Implementation Brief

* [x] In `packages/eslint-plugin/rules/acronym-case.js`
  * Expand the list of Acronyms to include the items listed in AC https://github.com/google/site-kit-wp/blob/3eacb691d9d832ac0f473f17e54c1bb581205a89/packages/eslint-plugin/rules/acronym-case.js#L108

### Test Coverage

* Include few new acronyms to the list of existing tests in `packages/eslint-plugin/rules/acronym-case.test.js`

## QA:Eng Brief

* Run unit tests for `acronym-case.js`.
* Confirm new acronyms (`SDK`, `EMM`, `REST`, `GKE`, `URI`, `UX`, `OS`, `GCP`, `DNS`, `FCM`, `IAM`, `IP`, `ISO`) are correctly flagged when casing is incorrect (e.g., `Sdk`).
* Verify new acronyms pass when casing is correct (e.g., `sdk`, `SDK_CONSTANT`).
* Ensure existing acronyms (e.g., `HTML`, `JSON`, `ID`) still behave as expected.
* Check that ignored cases (imports, function calls, global objects) continue to be ignored.

## Changelog entry

* N/A.

**Comment by bethanylang** [comment]
@tofumatt If you're not working on this one, can you please unassign so it can be picked up by someone else? Thanks!

**Comment by tofumatt** [comment]
@bethanylang Yup, sorry about that. Unassigned for now as I didn't get to this! 😅

**Comment by hussain-t** [comment]
### QA:Eng Verified ✅

* Verified that all newly added acronyms (SDK, EMM, REST, GKE, URI, UX, OS, GCP, DNS, FCM, IAM, IP, ISO) are correctly flagged when casing is incorrect (e.g., Sdk) and allowed when correct (e.g., SDK).
* Confirmed that existing acronyms such as HTML, JSON, and ID continue to behave correctly.
* Verified that ignored cases (e.g., global functions, imported names) are not incorrectly flagged.
